transmart-core-db-tests/test/integration/org/transmartproject/db/dataquery/highdim/assayconstraints/DefaultOntologyTermConstraintTests.groovy

```diff
@@ -42,13 +42,11 @@ class DefaultOntologyTermConstraintTests {
 
     @Test
     void basicTest() {
-        AssayQuery assayQuery = new AssayQuery([
+        List<AssayColumn> assays = DeSubjectSampleMapping.getOrderedAssaysDetachedCriteria([
                 new DefaultOntologyTermConstraint(
                         term: conceptsResourceService.getByKey('\\\\i2b2 main\\foo\\bar')
                 )
-        ])
-
-        List<AssayColumn> assays = assayQuery.retrieveAssays()
+        ]).list()
```

Comment:
I think getting rid of `AssayQuery` was a step backwards. The previous code was easier to read. You could just have made it implement `QueryableCriteria` and `Iterable`.